### PR TITLE
Fix initial test outputs across tracked tests - remove framework laden exceptions

### PR DESCRIPTION
**Pull request by stevendae**
### Ticket
Link to Github Issue: https://github.com/tenstorrent/tt-metal/issues/29063

### Problem description
Sweeps to Superset flow is executing reliably(see workflow [here](https://github.com/tenstorrent/tt-metal/actions/workflows/ttnn-run-sweeps.yaml)) however inspection of the sweep results on Superset show that many exceptions are occurring at the level of the test infrastructure. The implication being that we are not retrieving useful data about the op's operation with its passed parameters but rather that for the set of parameters the op test is not even being run. 

Some of these exceptions found include:

- improper input of parameters to functions (ttnn op, torch equivalent, or elsewhere)
- incompatible parameters
- xfail and xpass not being handled

Affected sweeps:
- reduction -> mean, prod, std, sum, var
- eltwise binary bcast
- eltwise binary fmod
- eltwise unary_backware pow_bw_sharded

### What's changed
This problem was resolved by reproducing each error/exception mentioned under the sweep run in the superset sweeps dashboard by running the op + vector combo that caused the exception. The reproduction was inspected and handled for each case based on values and types of variables being passed at the point of failure/exception.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by sjameelTT — APPROVED** [review]

File: tests/sweep_framework/sweeps/reduction/prod.py [hunk 1]
```diff
@@ -74,6 +62,10 @@ def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
     ):
         return True, "Row major is only supported for fp32 & fp16"
 
+    # ttnn.prod doesn't support keepdim=True when reducing over all dimensions (dim=None)
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/prod.py:65]: For the removal of shapes (`[0, 1], ...) above and this if, could you please create an issue and list it here.

The approach for sweeps earlier was to lay out various combinations, get a baseline passing rate, e.g. 20%, and then iterate and watch the passing rate improve. The approach has changed with the new framework to only adding tests that pass.

File: tests/sweep_framework/sweeps/reduction/sum.py [hunk 2]
```diff
@@ -37,25 +37,7 @@
...
-            [1, 3],
-            [2, 3],
-            [0, 1, 2],
-            [0, 1, 3],
-            [0, 1, 3],
-            [0, 2, 3],
-            [1, 2, 3],
-            [0, 1, 2, 3],
-        ],
+        "dim": [0, 1, 2, 3, None],
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/sum.py]: Did the other dims not work?
If not, please create an issue.

File: ttnn/ttnn/operations/reduction.py [hunk 3]
```diff
@@ -12,8 +12,14 @@ def golden_function(input_tensor: ttnn.Tensor, dim: Optional[Union[int, Tuple[in
         import torch
 
         torch_function = getattr(torch, torch_function_name)
-        if dim == None:
-            return torch_function(input_tensor, keepdim=keepdim)
+        if dim is None:
+            # When dim is None, PyTorch reduces over all dimensions
+            # For keepdim to work, we need to specify all dimensions explicitly
+            if keepdim:
+                all_dims = tuple(range(len(input_tensor.shape)))
```
Comment (bbradelTT) [inline ttnn/ttnn/operations/reduction.py:19]: dim = None and dim = [all values] should be two distinct tests.

If this wasn't working somewhere, an issue should be created.

**Review by patrickroberts — APPROVED** [review]

**Review by dchenTT — APPROVED** [review]

File: tests/sweep_framework/sweeps/reduction/prod.py (hunk 1 above)
Comment (stevendae) [reply tests/sweep_framework/sweeps/reduction/prod.py:65]: Reason why these are erased is because I referenced the [ttnn.prod documentation](https://docs.tenstorrent.com/tt-metal/latest/ttnn/ttnn/api/ttnn.prod.html) and found that the `dim` argument in this op does not accept list types. 

So I viewed this as applying an incompatible parameter to the op.

Looking at the `parameter` dict across many of the `reduction` sweeps I see that they are copies of each other. 

Including examples: prod_bw, mean, topk, std. 

The dimensions that were lists in which the op does not accept lists are removed from the sweep's parameters.

Torch documentation reflects this as well and tests with torch.prod.

File: tests/sweep_framework/sweeps/reduction/sum.py (hunk 2 above)
Comment (stevendae) [reply tests/sweep_framework/sweeps/reduction/sum.py]: torch equivalent accepts tuples of ints. so this will be added back except with invalidation measures to remove vectors with repeating dimensions when the dimension mentioned in a list is greater than the dimension of the input shape and must be reduced.

File: ttnn/ttnn/operations/reduction.py (hunk 3 above)
Comment (stevendae) [reply ttnn/ttnn/operations/reduction.py:19]: Not for the ops that use this golden function. For all the following ops
`mean, sum, max, min, var, std`
dim = None implies reducing across all dimensions.

**Review by stevendae — COMMENTED** [review]
@bbradelTT can you review my responses?

File: tests/sweep_framework/sweeps/reduction/prod.py (hunk 1 above)
Comment (edwinleeTT) [reply tests/sweep_framework/sweeps/reduction/prod.py:65]: @stevendae prod has two forms - there's one version that takes a `dim` int, and one that takes a `dims` list. I think both are bound together, so only one doc got provided for both. I'll update this doc to list both together.

I think these tests cover the `dims` case:
https://github.com/tenstorrent/tt-metal/blob/833d3d6c8b2ac1c1e918208735f8d6516d6d0d2b/tests/tt_eager/python_api_testing/unit_testing/test_prod_nc.py#L77

Comment (stevendae) [reply tests/sweep_framework/sweeps/reduction/prod.py:65]: Created an issue to add in sweeps that use list/tuple as dim. Torch does not have an implementation of prod that accepts list dims but ttnn does according to @edwinleeTT 

https://github.com/tenstorrent/tt-metal/issues/29844

**Review by bbradelTT — APPROVED** [review]

**Review by pavlejosipovic — APPROVED** [review]

File: .github/workflows/ttnn-run-sweeps.yaml [hunk 4]
```diff
@@ -556,12 +554,16 @@ on:
...
+        description: "Skip remaining tests in running suite after first timeout?"
+        type: boolean
+        default: true
       upload_results:
         description: "Upload sweep results to Superset?"
         type: boolean
         default: true
   schedule:
-    - cron: "0 4 * * 4" # This cron schedule runs the workflow at 4:00 AM UTC on Thursdays (Wednesday 11 PM EST)
+    - cron: "0 4 * * 1-5" # This cron schedule runs the workflow at 4:00 AM UTC on Monday-Friday
```
Comment (ayerofieiev-tt) [inline .github/workflows/ttnn-run-sweeps.yaml]: Why only Mon-Fri?

**Review by williamlyTT — APPROVED** [review]

File: .github/workflows/ttnn-run-sweeps.yaml (hunk 4 above)
Comment (stevendae) [reply .github/workflows/ttnn-run-sweeps.yaml]: Saturday morning is the comprehensive run

Comment (stevendae) [reply .github/workflows/ttnn-run-sweeps.yaml]: modifed to run everyday with comprehensive tests on Wednedsay and Saturday morning

**Review by aczajkowskiTT — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by blozano-tt — APPROVED** [review]
